packages/squiggle-lang/src/rescript/Distributions/PointSetDist/PointSetDist_Scoring.res

```diff
@@ -1,46 +1,150 @@
+    | (_, _) =>
+      getMixedSums(estimate, answer)->E.R2.fmap(((discretePart, continuousPart)) =>
+        discretePart +. continuousPart
+      )
+    }
+  }
+
+  let sumWithPrior = (
+    ~estimate: pointSetDist,
+    ~answer: pointSetDist,
+    ~prior: pointSetDist,
+    ~combineFn,
+    ~integrateFn,
+    ~toMixedFn,
+  ): result<score, Operation.Error.t> => {
+    let kl1 = sum(~estimate, ~answer, ~combineFn, ~integrateFn, ~toMixedFn)
+    let kl2 = sum(~estimate=prior, ~answer, ~combineFn, ~integrateFn, ~toMixedFn)
+    E.R.merge(kl1, kl2)->E.R2.fmap(((kl1', kl2')) => kl1' -. kl2')
+  }
 }
 
-module LogScoreWithPointResolution = {
-  let logFn = Js.Math.log
-  let score = (
-    ~priorPdf: option<float => float>,
-    ~predictionPdf: float => float,
-    ~answer: float,
-  ): result<float, Operation.Error.t> => {
-    let numerator = answer->predictionPdf
-    if numerator < 0.0 {
-      Operation.PdfInvalidError->Error
-    } else if numerator == 0.0 {
-      infinity->Ok
-    } else {
-      -.(
-        switch priorPdf {
-        | None => numerator->logFn
-        | Some(f) => {
-            let priorDensityOfAnswer = f(answer)
-            if priorDensityOfAnswer == 0.0 {
-              neg_infinity
-            } else {
-              (numerator /. priorDensityOfAnswer)->logFn
-            }
-          }
+module WithScalarAnswer = {
+  let sum = (mp: PointSetTypes.MixedPoint.t): float => mp.continuous +. mp.discrete
+  let score = (~estimate: pointSetDist, ~answer: scalar): result<score, Operation.Error.t> => {
+    let _score = (~estimatePdf: float => option<float>, ~answer: float): result<
+      score,
+      Operation.Error.t,
+    > => {
+      let density = answer->estimatePdf
+      switch density {
+      | None => Operation.PdfInvalidError->Error
+      | Some(density') =>
+        if density' < 0.0 {
+          Operation.PdfInvalidError->Error
+        } else if density' == 0.0 {
+          infinity->Ok
+        } else {
+          density'->logFn->(x => -.x)->Ok
         }
-      )->Ok
+      }
     }
+
+    let estimatePdf = x =>
+      switch estimate {
+      | Continuous(esti) => Continuous.T.xToY(x, esti)->sum->Some
+      | Discrete(esti) => Discrete.T.xToY(x, esti)->sum->Some
+      | Mixed(_) => None
+      }
+    _score(~estimatePdf, ~answer)
+  }
+
+  let scoreWithPrior = (~estimate: pointSetDist, ~answer: scalar, ~prior: pointSetDist): result<
+    score,
+    Operation.Error.t,
+  > => {
+    E.R.merge(score(~estimate, ~answer), score(~estimate=prior, ~answer))->E.R2.fmap(((s1, s2)) =>
+      s1 -. s2
+    )
   }
 }
+
+let twoGenericDistsToTwoPointSetDists = (~toPointSetFn, estimate, answer): result<
+  (pointSetDist, pointSetDist),
+  'e,
+> => E.R.merge(toPointSetFn(estimate, ()), toPointSetFn(answer, ()))
+
+let logScore = (args: scoreArgs, ~combineFn, ~integrateFn, ~toMixedFn): result<
+  score,
+  Operation.Error.t,
+> =>
+  switch args {
+  | DistEstimateDistAnswer({estimate, answer, prior: None}) =>
```

Comment:
I imagine these should no longer be called ``DistEstimate``, if all the estimates are dists